main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala

```diff
@@ -534,6 +534,37 @@ object ResolutionError {
     override def explain(formatter: Formatter): Option[String] = None
   }
 
+  /**
+    * An error indicating the number of term patterns does not match the enum tag term count.
+    *
+    * @param caze     the enum case symbol.
+    * @param expected the expected number of terms.
+    * @param actual   the actual number of patterns.
+    * @param loc      the location where the error occurred.
+    */
+  case class MismatchedTagPatternArity(caze: Symbol.CaseSym, expected: Int, actual: Int, loc: SourceLocation) extends ResolutionError {
+    override def summary: String = s"Expected ${Grammar.n_things(expected, "term")} but found $actual."
+
+    /**
+      * Returns the formatted error message.
+      */
+    override def message(formatter: Formatter): String = messageWithLink {
+      import formatter.*
+      s""">> Mismatched arity.
```

Comment:
-> Expected ${Grammar.n_things(expected, "argument")} for Cyan(caze) but found ${${Grammar.n_things(actual, "argument")}} arguments.